### PR TITLE
MM-33180: Extend Group APIs for plugins.

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Add `GetGroupMemberUsers` (mapping to `GetGroupMemberUsersPage`) and `GetGroupsBySource`.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-33180

#### Release Note
```release-note
Added GetGroupMemberUsers and GetGroupsBySource to the plugin API.
```

### CODE REVIEW

File: app/plugin_api.go [hunk 1]
```diff
@@ -536,6 +536,16 @@ func (api *PluginAPI) GetGroupByName(name string) (*model.Group, *model.AppError
...
 }
 
+func (api *PluginAPI) GetGroupMemberUsers(groupID string, page, perPage int) ([]*model.User, *model.AppError) {
+	users, _, err := api.app.GetGroupMemberUsersPage(groupID, page, perPage)
+
+	return users, err
+}
+
+func (api *PluginAPI) GetGroupsBySource(groupSource model.GroupSource) ([]*model.Group, *model.AppError) {
+	return api.app.GetGroupsBySource(groupSource)
```
Comment (lieut-data) [inline app/plugin_api.go:546]: There wasn't a paginated version of this, but I wasn't going to worry about it.

File: app/plugin_api.go [hunk 2]
```diff
@@ -536,6 +536,16 @@ func (api *PluginAPI) GetGroupByName(name string) (*model.Group, *model.AppError
 	return api.app.GetGroupByName(name, model.GroupSearchOpts{})
 }
 
+func (api *PluginAPI) GetGroupMemberUsers(groupID string, page, perPage int) ([]*model.User, *model.AppError) {
+	users, _, err := api.app.GetGroupMemberUsersPage(groupID, page, perPage)
```
Comment (lieut-data) [inline app/plugin_api.go:540]: Opted to only expose the paginated version vs. the "get all" given the potential size of groups (vs. the number of groups below).

**Review by crspeller — APPROVED** [review]

**Review by agarciamontoro — APPROVED** [review]
Looks good!

**Comment by lieut-data** [comment]
@prapti, possibly easiest to review this as part of https://github.com/mattermost/mattermost-plugin-incident-collaboration/pull/504?

**Review by prapti — APPROVED** [review]
Works as expected: tested and verified the test steps as mentioned iin https://github.com/mattermost/mattermost-plugin-incident-collaboration/pull/504.
Approving.

**Pull request merged**